store/src/java/com/zimbra/cs/event/logger/EventLogger.java

```diff
@@ -154,8 +159,16 @@ public void shutdownEventNotifierExecutor() {
         }
 
         ZimbraLog.event.warn("Shutdown called for Event Notifier Executor; initiating shutdown sequence...");
-        executorService.shutdownNow();
+        shutdownExecutor.set(true);
+        executorService.shutdown();
         try {
+            if(drainQueueBeforeShutdown.get()) {
+                for (int i = 0; i < config.getNumThreads(); i++) {
+                    /* In case of event queue being full we dont want to wait till space is available
+                    In that case some EventNotifier thread may not shutdown gracefully */
+                    eventQueue.offer(POISON_PILL, POISON_PILL_OFFER_TIMEOUT, TimeUnit.SECONDS);
+                }
+            }
             executorService.awaitTermination(30, TimeUnit.SECONDS);
```

Comment:
I know this isn't new but 30 seconds seems like a long time, would be nice if it was configurable.  Would this affect `zmcontrol stop`?  For instance, I think Zimbra is started and stopped several times during install.  Hope this wouldn't result in each of those taking up to 30 seconds?

Reply:
@fatkudu 30 seconds is the maximum time to wait before killing the EventNotifier threads. This time is needed to flush all the events in the logger to Solr as we are using a batch mechanism to commit events to the Solr.  

This would affect the `zmcontrol stop` process. The maximum time added by EventNotifier would be 30 seconds and minimum would be 3 seconds in case of an empty event queue.(after implementing the code change suggested by Travis). 

With the poison pill approach (current implementation), it could be max 30 sec and min would be time needed for all threads to consume the poison pill (~ 1 sec), again assuming the event queue is empty.

In case the event queue is not empty it could be anywhere between the max and min value.

In case of the `zimbra install` I would assume that event queue will be empty hence min time will be taken.